src/soundio/sounddeviceportaudio.cpp

```diff
@@ -248,11 +249,26 @@ SoundDeviceError SoundDevicePortAudio::open(bool isClkRefDevice, int syncBuffers
     m_syncBuffers = syncBuffers;
 
     // Create the callback function pointer.
-    PaStreamCallback* callback = nullptr;
+    PaStreamCallback* pCallback = nullptr;
     if (isClkRefDevice) {
-        callback = paV19CallbackClkRef;
+        pCallback = paV19CallbackClkRef;
+    } else if (framesPerBuffer == paFramesPerBufferUnspecified) {
+        m_syncBuffers = 1;
+        // This happens in case of JACK, where PortAudio creates artificial
+        // device streams from one native JACK server callback "JackCallback()"
+        // For every sound hardware. Clock drift can not happen, but the buffers
+        // are processed: In 1 - Out 1 - In 2 - Out 2
+        // This causes even with the "Experimental (no delay)" setting
+        // one extra buffer delay for the non clock reference device
+        pCallback = paV19Callback;
+        if (m_outputParams.channelCount) {
```

Comment:
please refrain from using the implicit conversion from int to `bool`. In the case when converting from double, clazy even warns of that.